genie-agent/src/main/java/com/netflix/genie/agent/execution/ExecutionAutoConfiguration.java

```diff
@@ -92,114 +88,44 @@
      */
     @Bean
     @Lazy
-    @ConditionalOnMissingBean
-    public LoggingListener loggingListener() {
+    @ConditionalOnMissingBean(LoggingListener.class)
+    public JobExecutionListener loggingListener() {
```

Comment:
You should make these the concrete classes as the return types. You don't need to downcast them for them to be picked up in collection injections and using the concrete classes lets them be recognized easier for conditionals.

Reply:
I'll change them, but `@ConditionalOnMissingBean` (without arguments) already does the right thing: create a bean of that concrete type if it doesn't already exist.